app/src/routes/v1/keycloak.js

```diff
@@ -23,4 +32,70 @@ keycloakRouter.get('/serviceClients', keycloak.protect('realm:GETOK_ADMIN'), asy
   }
 });
 
+// Creates a service client based on the configuration posted
+keycloakRouter.post('/configForm', [
+  body('configForm.applicationAcronym').isString(),
+  body('configForm.applicationName').isString(),
+  body('configForm.applicationDescription').isString(),
+  body('configForm.commonServices').isArray(),
+  body('configForm.clientEnvironment').isIn(['DEV', 'TEST', 'PROD']),
+  body('passwordPublicKey').isString()
+], async (req, res) => {
+  const userid = req.kauth.grant.access_token.content.sub;
+
+  // Validate for Bad Requests
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    log.debug('/configForm', 'Validation Error');
+    return res.status(400).json({
+      errors: errors.array(),
+      message: 'Validation failed'
+    });
+  }
+
+  const {
+    configForm,
+    passwordPublicKey: publicKey
+  } = req.body;
+
+  // Check for required permissions. Can only create clients for the acronyms you are associated with
+  const permissionErr = await permissionHelpers.checkAcronymPermission(userid, configForm.applicationAcronym);
+  if (permissionErr) {
+    return res.status(403).json({
+      message: permissionErr
+    });
+  }
+
+  // TODO: too much logic in the routing layer, refactor along with some other stuff that does KC client operations
+  // Group with the calls to "clients" component above and the one that's in Utils?
```

Comment:
Yea - we should make sure there's a backlog ticket somewhere to encapsulate this so we don't have this kind of code duplication balloon out of control. Their signatures are very similar.